src/test/ui/match/issue-72680.rs

```diff
@@ -0,0 +1,65 @@
+// run-pass
+
+#![feature(or_patterns)]
+
+fn main() {
+    assert_eq!(f("", 0), true);
+    assert_eq!(f("a", 1), true);
+    assert_eq!(f("b", 1), true);
+
+    assert_eq!(f("", 1), false);
+    assert_eq!(f("a", 0), false);
+    assert_eq!(f("b", 0), false);
```

Comment:
I think it would be clearer if written this way:
```rust
assert!(f("", 0));
assert!(f("a", 1));
assert!(f("b", 1));

assert!(!f("", 1)); // alternatively more explicit `.not()` may be used
assert!(!f("a", 0)); 
assert!(!f("b", 0));
```
(same for the `assert`s below)